crates/topos-p2p/src/behaviour/discovery.rs

```diff
@@ -69,7 +74,16 @@ impl DiscoveryBehaviour {
         Self {
             inner: kademlia,
             current_bootstrap_query_id: None,
-            next_bootstrap_query: Some(Box::pin(tokio::time::interval(config.bootstrap_interval))),
+            next_bootstrap_query: if known_peers.is_empty() {
```

Comment:
The `know_peers` could be either bootnodes from `genesis` file or defined in config.

The `next_bootstrap_query` is the inteval trigger of the `bootstrap` query used in the `poll` function. For `genesis` bootnode I encounter the fact that if they're alone at start, they can't send any `bootstrap` query because they know no one. As a quick win I just disable the `bootstrap` queries for lonely bootnode, meaning that they will not try to walk the network but still receive connection and aggregate those connection.

This is only possible for `bootnode` as the `known_peers` have the list of bootnode from the genesis file, during the configuration parse, the local peer_id is excluded from this list. If the local node is the only bootnode in the genesis file, the list is then empty but for other node nothing change.

---

> Both of next_bootstrap_query and health_status are confusing to me. No known peers means we're healthy?

`next_bootstrap_query` is `The next bootstrap query interval used to schedule the next bootstrap query` as define above

`health_status` is the current heath status of the behaviour based on the event it consumes.

For a lonely bootnode that doesn't need any discovery the `HealthStatus` is `healthy` by default. For any other node the healthy status is reached when the first `bootstrap` query to resolve with one of the defined `bootnode` are successful.

Reply:
Cool, thank you for explaining, much clearer now. Can you add a comment explaining that "Bootnodes do not have any known peers, so we skip bootstrapping for them" or something similar?